torch/utils/data/dataset.py

```diff
@@ -260,16 +260,25 @@ def __len__(self):
         return len(self.indices)
 
 
-def random_split(dataset, lengths):
+def random_split(dataset, lengths, random_state=None, generator=None):
```

Comment:
Any time you wrote `random_split(d, l, state)`, I could write instead `random_split(d, l, torch.Generator.manual_seed(state))`. The Pythonic way is that there is one obvious way to do it. So I suggest getting rid of the `random_state` argument entirely.

Reply:
Oh I hadn't realised `manual_seed` returns the generator so it's possible to keep a one-liner.